tests/sqlAst/data/sql-ast-narrowing.php

```diff
@@ -7,6 +7,15 @@
 
 class Foo
 {
+    public function noFromTable(PDO $pdo): void
```

Comment:
does the change also make type inference work for cases where the FROM part is a subselect or similar? if so, would be great to have a test-case for it

Reply:
No, it does not handle `TableReferenceSubquery` for the FROM clause yet. I kind of see a way to do it by recursively using `QueryReflection::getResultType`, but the `TableReferenceSubquery` class doesn't retain the subquery string, so I think we'd possibly need some changes to SqlFtw to make it happen.

In the meantime, I'll make the logic in this PR more explicit so that it's clear it still does not yet support subqueries.